### PR TITLE
Add new patch operation for user profile settings

### PR DESCRIPTION
**Pull request by sigridbra**
## Related Issue(s)
- #605 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

* **New Features**
  * New PATCH endpoint to partially update current user profile settings (language, party preference, unit/sub-entity visibility, preselected party).

* **Improvements**
  * Safer handling and serialization of optional/null profile fields, including explicit support for optional values.

* **Tests**
  * Added integration and unit tests covering PATCH behavior, edge cases and optional serialization.

* **Chores**
  * Updated runtime JSON serialization configuration and profile integration settings.

### CODE REVIEW

File: src/Altinn.Profile.Core/Utils/OptionalJsonConverter.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Altinn.Profile.Core.Utils
+{
+    /// <summary>
+    /// Converts <see cref="Optional{T}"/> values to and from JSON, handling explicit nulls and missing values.
+    /// </summary>
+    /// <typeparam name="T">The type of the value contained in the <see cref="Optional{T}"/>.</typeparam>
+    public class OptionalJsonConverter<T> : JsonConverter<Optional<T>>
```
Comment (SandGrainOne) [inline src/Altinn.Profile.Core/Utils/OptionalJsonConverter.cs]: Could we try to create a few unit tests on this or does the controller tests cover everything?

File: test/Bruno/environments/Local.bru [hunk 2]
```diff
@@ -1,10 +1,10 @@
 vars {
-  Environment: dev
+  Environment: at22
```
Comment (SandGrainOne) [inline test/Bruno/environments/Local.bru]: Revert these changes. Though we do need a better method of testing port forwarded endpoints.
